BuildResidentialScheduleFile/resources/schedules.rb

```diff
@@ -0,0 +1,1048 @@
+      transition_matrix_file_weekend = args[:resources_path] + "/schedules_weekend_mkv_chain_transition_prob_cluster_#{occ_type_id}.csv"
+      transition_matrix_weekend = CSV.read(transition_matrix_file_weekend)
+      transition_matrix_weekend = transition_matrix_weekend.map { |x| x.map { |y| y.to_f } }
+
+      simulated_values = []
+      @total_days_in_year.times do |day|
+        today = @sim_start_day + day
+        day_of_week = today.wday
+        if [0, 6].include?(day_of_week)
+          # Weekend
+          day_type = 'weekend'
+          initial_prob = initial_prob_weekend
+          transition_matrix = transition_matrix_weekend
+        else
+          # weekday
+          day_type = 'weekday'
+          initial_prob = initial_prob_weekday
+          transition_matrix = transition_matrix_weekday
+        end
+        j = 0
+        state_prob = initial_prob # [] shape = 1x7. probability of transitioning to each of the 7 states
+        while j < @mkc_ts_per_day do
+          active_state = weighted_random(prng, state_prob) # Randomly pick the next state
+          state_vector = [0] * 7 # there are 7 states
+          state_vector[active_state] = 1 # Transition to the new state
+          # sample the duration of the state, and skip markov-chain based state transition until the end of the duration
+          activity_duration = sample_activity_duration(prng, activity_duration_prob_map, occ_type_id, active_state, day_type, j / 4)
+          activity_duration.times do |repeat_activity_count|
+            # repeat the same activity for the duration times
+            simulated_values << state_vector
+            j += 1
+            if j >= @mkc_ts_per_day then break end # break as soon as we have filled acitivities for the day
+          end
+          if j >= @mkc_ts_per_day then break end # break as soon as we have filled activities for the day
+
+          transition_probs = transition_matrix[(j - 1) * 7...j * 7] # obtain the transition matrix for current timestep
+          state_prob = transition_probs[active_state]
+        end
+      end
+      # Markov-chain transition probabilities is based on ATUS data, and the starting time of day for the data is
+      # 4 am. We need to shift everything forward by 16 timesteps to make it midnight-based.
+      simulated_values = simulated_values.rotate(-4 * 4) # 4am shifting (4 hours  = 4 * 4 steps of 15 min intervals)
+      all_simulated_values << Matrix[*simulated_values]
+    end
+    # shape of all_simulated_values is [2, 35040, 7] i.e. (geometry_num_occupants, period_in_a_year, number_of_states)
+    plugload_sch = schedule_config['plugload']
+    lighting_sch = schedule_config['lighting']
+    ceiling_fan_sch = schedule_config['ceiling_fan']
+
+    monthly_lighting_schedule = schedule_config['lighting']['monthly_multiplier']
+    holiday_lighting_schedule = schedule_config['lighting']['holiday_sch']
+
+    sch = Lighting.get_schedule(@model, @epw_file)
```

Comment:
It looks like Lighting.get_schedule doesn't actually use the `model`. Can you remove this argument and update all references?